crates/rattler_shell/src/activation.rs

```diff
@@ -459,21 +466,107 @@ impl<T: Shell + Clone> Activator<T> {
 
         script.set_path(path.as_slice(), variables.path_modification_behavior)?;
 
-        // deliberately not taking care of `CONDA_SHLVL` or any other complications at
-        // this point
+        // Get the current shell level
+        let shlvl = variables
+            .current_env
+            .get("PIXI_SHLVL")
+            .and_then(|s| s.parse::<i32>().ok())
+            .unwrap_or(0);
+
+        // Save original CONDA_PREFIX value if it exists
+        if let Some(existing_prefix) = variables.current_env.get("CONDA_PREFIX") {
+            script.set_env_var(
+                &format!("PIXI_ENV_SHLVL_{shlvl}_CONDA_PREFIX"),
+                existing_prefix,
+            )?;
+        }
+
+        // Set new CONDA_PREFIX
         script.set_env_var("CONDA_PREFIX", &self.target_prefix.to_string_lossy())?;
 
+        // For each environment variable that was set during activation
         for (key, value) in &self.env_vars {
+            // Save original value if it exists
+            if let Some(existing_value) = variables.current_env.get(key) {
+                script.set_env_var(&format!("PIXI_ENV_SHLVL_{shlvl}_{key}"), existing_value)?;
+            }
+            // Set new value
             script.set_env_var(key, value)?;
         }
 
+        // Set the new PIXI_SHLVL
+        script.set_env_var("PIXI_SHLVL", &(shlvl + 1).to_string())?;
```

Comment:
I think we need to move setting the env var on top, otherwise we will never set first PIXI_SHLV_0 ( because if it is not set, we take it 0 and make it + 1 )